bootstrap/spigot/src/main/java/org/geysermc/geyser/platform/spigot/GeyserSpigotPlugin.java

```diff
@@ -165,31 +163,38 @@ public void onGeyserInitialize() {
 
     @Override
     public void onEnable() {
-        this.geyserCommandManager = new GeyserSpigotCommandManager(geyser);
-        this.geyserCommandManager.init();
-
-        // Because Bukkit locks its command map upon startup, we need to
-        // add our plugin commands in onEnable, but populating the executor
-        // can happen at any time (later in #onGeyserEnable())
-        CommandMap commandMap = GeyserSpigotCommandManager.getCommandMap();
-        for (Extension extension : this.geyserCommandManager.extensionCommands().keySet()) {
-            // Thanks again, Bukkit
-            try {
-                Constructor<PluginCommand> constructor = PluginCommand.class.getDeclaredConstructor(String.class, Plugin.class);
-                constructor.setAccessible(true);
-
-                PluginCommand pluginCommand = constructor.newInstance(extension.description().id(), this);
-                pluginCommand.setDescription("The main command for the " + extension.name() + " Geyser extension!");
+        // Create command manager early so we can add Geyser extension commands
+        var sourceConverter = new CommandSourceConverter<>(
+                CommandSender.class,
+                Bukkit::getPlayer,
+                Bukkit::getConsoleSender,
+                SpigotCommandSource::new
+        );
+        LegacyPaperCommandManager<GeyserCommandSource> cloud;
+        try {
+            // LegacyPaperCommandManager works for spigot too. todo: use PaperCommandManager instead for Paper 1.20.6+
+            cloud = new LegacyPaperCommandManager<>(
+                    this,
+                    ExecutionCoordinator.simpleCoordinator(),
+                    sourceConverter
+            );
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
 
-                commandMap.register(extension.description().id(), "geyserext", pluginCommand);
-            } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException ex) {
-                this.geyserLogger.error("Failed to construct PluginCommand for extension " + extension.name(), ex);
+        if (cloud.hasCapability(CloudBukkitCapabilities.BRIGADIER)) {
```

Comment:
Apparently, we're supposed to check for `CloudBukkitCapabilities.NATIVE_BRIGADIER` instead, atleast according to the cloud docs: https://cloud.incendo.org/minecraft/paper/#brigadier. No clue how these are different though.

Reply:
i think it says that because it's on the Paper page? not sure. side note, the javadocs seem oudated.

i looked at the method and fortunately it will still attempt commodore brigadier (which supports 1.13.2 - 1.18.2) if NATIVE_BRIGADIER is unavailable. i'm pretty sure that since we only support 1.13.2+, the only time this will throw is if the user is on Spigot 1.19+.

i'm just going to remove the if-guard because it doesn't matter and basically makes this easier for us